dockerfiles/docker-compose-search.yml

```diff
@@ -8,12 +8,12 @@ volumes:
 services:
 
   celery:
-    environment:
-      - SEARCH=t
+    command: ["../../docker/wait-for-it.sh", "search:9200", "--timeout=360", "--strict", "--", "../../docker/celery.sh"]
 
   web:
-    environment:
-      - SEARCH=t
+    # This appends two blocking wait-for-it.sh calls to one another and if both
+    # are successful, runs the web.sh entrypoint
+    command: ["../../docker/wait-for-it.sh", "search:9200", "--timeout=360", "--strict", "--", "../../docker/wait-for-it.sh", "database:5432", "--timeout=60", "--strict", "--", "../../docker/web.sh"]
```

Comment:
This is getting too complex, IMO. I don't really understand how these chained wait-for-it works and how it interacts with the `--no-search`.

Besides, we are removing the `SEARCH=t` variable from here which is required by the application at https://github.com/readthedocs/readthedocs.org/blob/fe0c723c9bfb7f2330b4f064ebbc2de76a807911/readthedocs/settings/docker_compose.py#L62

In fact, we _don't really require ES to be available **before** starting Celery or Web_. The only case when this is needed is when there is a Celery task from a previous execution. Web container doesn't require ES to startup... it's only required when the user performs a search.

At this point, I think it's better and simple to revert this change and use the `wait_for_search.py` script we used to have. I'm not opposed to this change, but we can introduce wait-for-it in another PR. For now, let's keep this PR scoped for the change of `mc` instead of `aws` which is what we want immediately.

Reply:
I can easily change this to not affect the entrypoint command -- it was more of an attempt to bring the behavior out in the docker compose configuration - so have it wait for other containers in the same layer that configures these containers.

Reply:
....but looking at the end result, I agree that the chained wait-for-it command wasn't very nice.

I completely missed how SEARCH was used further down in the stack. Thanks for pointing that out :+1: